File: 3718-smallest-missing-multiple-of-k/solution.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    int missingMultiple(std::vector<int>& nums, int k) {
        // Store all elements of nums in a hash set for O(1) lookup time
        std::unordered_set<int> numSet(nums.begin(), nums.end());
        
        // Start checking from the first positive multiple of k
        int multiple = k;
        
        // Increment by k until we find a multiple not present in the set
        while (numSet.count(multiple)) {
            multiple += k;
        }
        
        return multiple;
    }
};

File: 3718-smallest-missing-multiple-of-k/solution.py
class Solution:
    def missingMultiple(self, nums: list[int], k: int) -> int:
        # Convert list to set for O(1) average time lookup
        num_set = set(nums)
        
        # Start checking from the first positive multiple of k
        multiple = k
        
        # Increment by k until a missing multiple is found
        while multiple in num_set:
            multiple += k
            
        return multiple